### PR TITLE
Document and expose core.internal.dassert

### PR DESCRIPTION
**Pull request by Geod24**
Give this module a proper documentation to make it easier to maintain.

### CODE REVIEW

File: src/core/internal/dassert.d [hunk 1]
```diff
@@ -1,18 +1,62 @@
...
-Provides light-weight formatting utilities for pretty-printing
-on assertion failures
-*/
+ * Support for rich error messages generation with `assert`
+ *
+ * This module provides the `_d_assert_fail` hooks which are instantiated
+ * by the compiler whenever `-checkaction=context` is used.
+ * There are two hooks, one for unary expressions, and one for binary.
+ * When used, the compiler will rewrite `assert(a >= b)` as
+ * `assert(a >= b, _d_assert_fail!"<"(a, b))`, inverting the operator.
```
Comment (MoonlightSentinel) [inline src/core/internal/dassert.d:8]: dmd supplies the original operator which is inverted by `invertCompToken`.

```suggestion
 * `assert(a >= b, _d_assert_fail!">="(a, b))`.
```

File: src/core/internal/dassert.d [hunk 2]
```diff
@@ -1,18 +1,62 @@
...
 module core.internal.dassert;
 
-/// Allows customized assert error messages for unary expressions
+/**
+ * Generates rich assert error messages for unary expressions
+ *
+ * The unary expression `assert(!una)` will be turned into
+ * `assert(!una, _d_assert_fail!"!"(una))`.
+ * This routine simply forward to the binary version as if the
+ * user wrote `assert(una == false)`.
```
Comment (MoonlightSentinel) [inline src/core/internal/dassert.d:28]: It doesn't call `_d_assert_fail` but rather produces the same string as `_d_assert_fail!"=="(una, false)`.

File: src/core/internal/dassert.d [hunk 3]
```diff
@@ -1,18 +1,62 @@
...
     enum token = op == "!" ? "==" : "!=";
     return combine(val, token, "true");
 }
 
-/// Allows customized assert error messages for binary expressions
+/**
+ * Generates rich assert error messages for binary expressions
+ *
+ * The binary expression `assert(x == y)` will be turned into
+ * `assert(x == y, _d_assert_fail!"!="(x, y))`.
```
Comment (MoonlightSentinel) [inline src/core/internal/dassert.d:50]:
```suggestion
 * `assert(x == y, _d_assert_fail!"=="(x, y))`.
```

File: src/core/internal/dassert.d [hunk 4]
```diff
@@ -1,18 +1,62 @@
...
+ *
+ * Params:
+ *   comp = Invert of the comparison operator that was used in the expression.
```
Comment (MoonlightSentinel) [inline src/core/internal/dassert.d:53]:
```suggestion
 *   comp = comparison operator that was used in the expression.
```

**Comment by Geod24** [comment]
Thanks @MoonlightSentinel , updated according to your comments.

**Review by MoonlightSentinel — APPROVED** [review]

**Pull request merged**